planet/api/order_details.py

```diff
@@ -0,0 +1,539 @@
+# Copyright 2020 Planet Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+"""Functionality for preparing order details for use in creating an order"""
+from __future__ import annotations  # https://stackoverflow.com/a/33533514
+import copy
+import json
+from typing import List
+
+from .. import specs
+
+
+class OrderDetailsException(Exception):
+    """Exceptions thrown by OrderDetails"""
+    pass
+
+
+class OrderDetails():
```

Comment:
How do you plan to keep classes like these up to date with changing API specs?

Reply:
Manual updates that require a whole release cycle. Adding or changing the name of Items/Product Bundles will be easy to update via a list at the top. Anything else will require us to get our hands dirty and change the class associated with whatever change happens in the API. It's a pain, but honestly the API shouldn't change much. If it does, it either brakes our client or breaks the massive JSON blocks all the users of the client would have to build if we didn't have these classes. Do you have any thoughts on making this easier on us and users?

Reply:
One option is to generate classes from the swagger specs with a tool such as https://github.com/swagger-api/swagger-codegen

You have more flexibility and room for simplified abstractions by defining your own classes, though.

Reply:
My main concern can be expressed as wanting a one liner to place an order if I have an AOI and know which assets/item_types I want.

Reply:
**Thank you** for making me aware of this amazing tool! I am checking it out further. I'm also with you on wanting a one-liner. Is there any chance you could provide an ideal usage example?

Reply:
looking into codegen: #254, one-liner: #250

Reply:
looking into codegen: #254, one-liner: #250